### DISCUSSION TITLE
Redundant packet batching logic in usbd_control.c

### DISCUSSION THREAD

**Discussion by rianhunter** [hathach/tinyusb#1794] [General]
I noticed that calling `_data_stage_xact()` with a buffer larger than `CFG_TUD_ENDPOINT0_SIZE` will cause a series of calls to `usbd_edpt_xfer()` until the entire buffer is transferred.

Batching at this layer does not seem necessary since `usbd_edpt_xfer()` will handle large buffers. It seems you can change:

```
  uint16_t const xact_len = tu_min16(_ctrl_xfer.data_len - _ctrl_xfer.total_xferred, CFG_TUD_ENDPOINT0_SIZE);
```

into

```
  uint16_t const xact_len = _ctrl_xfer.data_len - _ctrl_xfer.total_xferred;
```

and everything will continue to work, the dcd layer will be responsible for splitting up the buffer into sequential packets.

I can provide a patch for that if you agree with my logic. Thank you

**Comment by hathach** [comment]
have you run the stack with/without the changes and observe its changed behavior ? If yes, then please  attach the txt containing logs file in both cases with highlight/note where the changes code apply/affect the behavior.
